### ISSUE TITLE
Sometimes thermostat switches off without reason

### ISSUE THREAD

**Issue by Bergoglio** [jmcollin78/versatile_thermostat#84]
I created and settled 4 Versatile thermostats (over 4 standard Home Assistant thermostats) and I don’t know why one of them (always the same) it switches off by itself, the others never.

**Comment by jmcollin78** [comment]
Hello @Bergoglio ,

if the thermostat switch off (I think you mean, goes to HVAC_OFF mode), it could be because the associated window sensor is detected as open. 

To be sure, you should diagnose what happens using the "Developpers Tools / State" accessible here:
![Capture d’écran 2023-06-07 à 22 58 56](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/3763f9aa-cb25-4df0-bc80-9c4405d1c688)
 and then:
![Capture d’écran 2023-06-07 à 22 59 36](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/2e043ecb-ca67-4926-adee-efde865142d4)

(type your climate entity name). Please give me what you see in the state. This is something looking like this:
 
![Capture d’écran 2023-06-07 à 23 00 36](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/afebb711-3839-41ba-ad76-cb86b6eaa948)

Maybe I will be able to help.

**Comment by Bergoglio** [comment]
Ok, I made some hardware tests (open the window, close the window, etc.)
Yes, I think Your idea is right : it goes off when the window is open.
And the problem can be the same in issue 85 : states don't refresh

This is in developer tools

![Screenshot 2](https://github.com/jmcollin78/versatile_thermostat/assets/64399503/8bb7a727-cbcd-4595-b94d-565d4e3da156)

and this what is inside

`hvac_modes:
  - heat
  - 'off'
min_temp: 7
max_temp: 35
target_temp_step: 0.1
preset_modes:
  - none
  - eco
  - comfort
  - boost
current_temperature: 22.8
temperature: 19.5
hvac_action: idle
preset_mode: eco
hvac_mode: heat
type: thermostat_over_climate
eco_temp: 19.5
boost_temp: 23
comfort_temp: 22
eco_away_temp: 19
boost_away_temp: 23
comfort_away_temp: 21
power_temp: null
target_temp: 19.5
current_temp: 22.8
ext_current_temperature: 24.2
current_power: null
current_power_max: null
saved_preset_mode: eco
saved_target_temp: 19.5
saved_hvac_mode: heat
window_state: 'on'
motion_state: null
overpowering_state: null
presence_state: home
window_auto_state: false
security_delay_min: 60
security_min_on_percent: 0.7
security_default_on_percent: 0.25
last_temperature_datetime: '2023-06-08T00:25:41.352046+02:00'
last_ext_temperature_datetime: '2023-06-08T00:24:04.550056+02:00'
security_state: false
minimal_activation_delay_sec: 300
device_power: 0.1
mean_cycle_power: null
total_energy: 0.40027724511111107
last_update_datetime: '2023-06-08T00:25:41.361355+02:00'
timezone: Europe/Rome
window_delay_sec: 30
window_auto_open_threshold: null
window_auto_close_threshold: null
window_auto_max_duration: null
underlying_climate: climate.bagno_caldo_nonna
start_hvac_action_date: null
friendly_name: Riscaldamento bagno Nonna
supported_features: 17
`

this is the screenshot of what it shows

![Screenshot 1](https://github.com/jmcollin78/versatile_thermostat/assets/64399503/3e009a25-e8bf-477c-87b7-07305228c4f7)

Pay attention : window is closed and Versatile tells open (Aperto/a) , my mother is at home and Versatile tells outside......

**Comment by Bergoglio** [comment]
To be more precise, the screenshots of the two sensors :

Window

![Screenshot 3](https://github.com/jmcollin78/versatile_thermostat/assets/64399503/23f18aaa-01a3-408f-9f24-c125a32c8df7)

Person

![Screenshot 4](https://github.com/jmcollin78/versatile_thermostat/assets/64399503/62dd69f9-6292-479b-b1a8-dda1be768a46)

**Comment by Bergoglio** [comment]
I made another test on window sensor and now I understand a problem : my sensor is on when the window is closed and off when the window is open (exactly the opposite of what Versatile wants....)
So, no problems on window detection.

**Comment by Bergoglio** [comment]
Solved !
My problem was the wrong sensor for window open detection.

I'm sorry.....

**Comment by jmcollin78** [comment]
No problem. This is first time I see a window sensor working at opposite. What is this window sensor ?

**Comment by Bergoglio** [comment]
The window sensor is a simple magnetic contact (brand Siegenia) in the window hardware (it report not only if the window is open or closed but also if only the window handle is open).
It is a normally closed contact so, if a thief localizes the cable and cut it an allarm can start to sound.
I pass trough this contact a 24V AC that work on a NO (normally open) relé (the contact will be normally closed until You open the window / handle). In the relé contact I use a 220V AC that activate a switch input (switch_0) in a Shelly Pro 4PM (the same that control fan speed in room's fancoil). I made in this way because Shelly devices accept only 220V and I don't like to pass 220V in Siegenia sensor cable (2 x 0,22 mmq).
Knowing that Versatile wants an on state when the window is open, I will use the 24V / 220 V relé not normally open but normally closed.
Now I solved making in configuration.yaml this

`template:
  - binary_sensor:
      - name: "finestra bagno nonna opposto"
        state: "{{ is_state('binary_sensor.shellypro4pm_30c6f7838614_switch_0_input', 'off') }}"`

and set in Versatile this entity

![Screenshot 1](https://github.com/jmcollin78/versatile_thermostat/assets/64399503/0cb05391-16fb-4b2c-a76f-b0372d40dbc7)
